lib/rules/lint-no-obsolete-elements.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const Rule = require('./base');
+
+const errorMessage = element => `Use of <${element}> detected. Do not use deprecated elements.`;
+
+const OBSOLETE_ELEMENTS = [
+  'applet',
+  'acronym',
+  'bgsound',
+  'dir',
+  'frame',
+  'frameset',
+  'noframes',
+  'isindex',
+  'keygen',
+  'listing',
+  'menuitem',
+  'nextid',
+  'noembed',
+  'plaintext',
+  'rb',
+  'rtc',
+  'strike',
+  'xmp',
+  'basefont',
+  'big',
+  'blink',
+  'center',
+  'font',
+  'marquee',
+  'multicol',
+  'nobr',
+  'spacer',
+  'tt',
+];
+
+module.exports = class NoObsoleteElements extends Rule {
+  visitor() {
+    return {
+      ElementNode(node) {
+        if (OBSOLETE_ELEMENTS.includes(node.tag)) {
```

Comment:
We also need to check if the `node.tag` is a block param.

Can you add a "good" test for:

```hbs
{{#let (component 'whatever-here') as |plaintext|}}
  <plaintext />
{{/let}}
```

^ breaks the `no-shadowed-elements` rule, but should **not** fail this rule

Reply:
@rwjblue just to be clear- we both want to prevent let-as-(obsolete element), right?

Reply:
Kinda, yes. _This_ rule should **not** flag any block params at all, but the no-shadowed-elements rule prevents using _any_ lowercased block param names in the element position. 

So under the no-shadowed-element rule (which is enabled by default), the example I gave above would be disallowed.

Reply:
Hmmm. I think it should block use of these elements, block params or not. Let's schedule a call to discuss more, so I can understand the logic.

Reply:
Seems good, but we won't be landing this rule until we have tests ensuring that this example does not fail the new rule being added:

```hbs
{{#let (component 'whatever-here') as |plaintext|}}
  <plaintext />
{{/let}}
```

Reply:
I see. I still would like to understand it better. Thanks! :+1: